Remove "Resolved By User" Relationship when WorkItem is reactivated

SimonZein opened the pull request:
When the Undo-WorkItemResolution is called, the Resolved By User Relationship remains.
Therefor additional code is added to the function

SimonZein committed:
Update smletsExchangeConnector.ps1

SimonZein committed:
Update smletsExchangeConnector.ps1

AdhocAdam commented:
Been doing some testing of this and it certainly works as expected. I had no doubt 😄.

And I was considering simplifying the catch statement down to a single line by removing
```powershell
$logMessage = $_.Exception.Message
```

and then just writing Exception into the event directly. But then it got me thinking, _"What are the conditions the CATCH would even engage?"_ So for the sake of argument I proceded to:
- Invoke Undo-WorkItemResolution on a recently created Incident/Problem (the relationship doesn't exist/is null)
- Resolve an Incident/Problem, then invoke Undo-WorkItemResolution (create 1 rel, get 1 rel, break 1 rel)
- Resolve an Incident/Problem, then Undo-WorkItemResolution, Resolve it via a different User, then Undo-WorkItemResolution (create 1 rel, break 1 rel, get 2 rels, break 1/the active rel). I was more curious about this one because SCSM keeps a history of the relationships EVEN if they aren't active any more via the _IsDeleted_ boolean. In any case, this would return _multiple_ relationships passed to a single Remove-SCSMRelationshipObject

And to my surprise, the CATCH will never engage in **_any_** of these scenarios. Nor will it produce **_any_** output. In fact, the only time the CATCH will engage is if the Work Item doesn't exist/cant be found. Which all things considered, should be impossible to get here with a Work Item that doesn't exist because the function Update-WorkItem is only way in with a total of 5 references within that function via Incidents/Problems. So that said, while the try/catch certainly would have been my default line of thinking as well, it appears it doesn't need to be used at all! Which means your suggestion can actually be executed in a single line:

```powershell
Get-SCSMRelationshipObject -BySource $workItem -Filter "RelationshipId -eq '$($workResolvedByUserRelClass.Id)'" @scsmMGMTParams | Remove-SCSMRelationshipObject
```

It's also worth adding, I think the reason I never thought to test this scenario until now is because **Remove-SCSMRelationshipObject** is not used **_anywhere_** in the connector.

But before any changes are made to your branch/this pull request, I figured I'd put this here first just to bounce it off you in the event there is something I'm overlooking or if your experience is different.

SimonZein commented:
Hello Adam,
nice to hear this one works as expected.
Yes the thing with the relationships...In my scripts, and I really don't know why I didn't add it to my function here, I always pipeline the relationships via

`| ? {$_.IsDeleted -eq $false}`

Maybe this is something worth adding, so even if the Get... returns more than one relationship, it would delete the actual one :)

AdhocAdam committed:
Remove-SCSMRelationshipObject one liner

After testing it does not appear there is a scenario wherein the try/catch will engage.
